### ISSUE TITLE
Latest docker image "500 Internal Server Error"

### ISSUE THREAD

**Issue by hoover900** [RikudouSage/LemmySchedule#16]
Just pulled the latest docker image and I'm getting a "500 Internal Server Error" when I try to schedule a post. going to roll back for the time being

not too sure going about saying what all version the image is an all, but portainer gives me a checksum
sha256:0279b1ccdb9095a0f83cca1dcc40d19045d2db2451ac9286f47284185aa4e9d0
for the current image I'm using

**Comment by hoover900** [comment]
okay figured out how to find it image version. the image I'm currently using is 1.3.3, which is borked for trying to schedule posts

**Comment by hoover900** [comment]
```
192.168.1.83 - - [16/Oct/2023:01:32:36 +0000] "GET /post/create HTTP/1.1" 500 1353 "http://unas.hanlan:8074/post/list" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/118.0"
192.168.1.83 - - [16/Oct/2023:01:32:35 +0000] "GET /post/list HTTP/1.1" 200 4279 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/118.0"
[Mon Oct 16 01:32:36.772882 2023] [php:notice] [pid 45] [client 192.168.1.83:3122] [critical] Uncaught PHP Exception Twig\\Error\\RuntimeError: "An exception has been thrown during the rendering of a template ("Unable to generate a URL for the named route "app.post.ajax.title_expression" as such route does not exist.")." at /var/www/html/templates/post/create.html.twig line 10, referer: http://unas.hanlan:8074/post/list
192.168.1.83 - - [16/Oct/2023:01:32:35 +0000] "GET /build/fonts/fa-solid-900.1551f4f6.woff2 HTTP/1.1" 200 78557 "http://unas.hanlan:8074/build/app.195b04ee.css" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/118.0"
192.168.1.83 - - [16/Oct/2023:01:32:35 +0000] "GET /build/app.195b04ee.css HTTP/1.1" 200 57083 "http://unas.hanlan:8074/post/list" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/118.0"
192.168.1.83 - - [16/Oct/2023:01:32:35 +0000] "GET /build/app.2fd1a0c3.js HTTP/1.1" 200 9253 "http://unas.hanlan:8074/post/list" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/118.0"
```

**Comment by hoover900** [comment]
so rolling the docker image back does not resolve the not being able to schedule issue. other than that scheduled posts seem to still be posting an all.

**Comment by hoover900** [comment]
k, figured out there's something in my cache that isn't playing nice since updating to 1.3.3. if I nuke my cache it resolves this issue. however this isn't ideal at the moment because I have several days of posts lined up. once my posts finish posting I'm just going to nuke the cache. 

let me know if you want the backup of the cache I made to poke at.

**Comment by RikudouSage** [comment]
Can you send the cache?

There should be a directory called `pools`, this is the one that contains your data and should be preserved, can you try deleting the others (keep a backup of the whole directory, though)?

**Comment by hoover900** [comment]
copy of my cache
[cache.zip](https://github.com/RikudouSage/LemmySchedule/files/12924041/cache.zip)

**Comment by hoover900** [comment]
deleting everything but the pools folder works

**Comment by RikudouSage** [comment]
I'm failing to reproduce the issue, was it perhaps caused by the stale cache?

**Comment by RikudouSage** [comment]
Closing the issue, feel free to reopen if the issue still persists.

**Comment by timnunes** [comment]
I'm having the same problem with a fresh installation.

The problem seems to only happen with recurring _weekly_ posts. I can change the time without problem, but if I try to change the day of the week, I get the pop-up error.

I _was_ able to schedule a weekly post on Monday, because Monday is filled in by default and there's no error if you don't change it.

**Comment by timnunes** [comment]
> Closing the issue, feel free to reopen if the issue still persists.

And here are some relevant logs:

```
[critical] Uncaught PHP Exception Exception: "Unknown or bad format (P-4D)" at /var/www/html/src/Service/ScheduleExpressionParser.php line 91, referer: http://myaddress:5823/post/create

myipaddress - - [24/Oct/2023:00:50:26 +0000] "GET /en/_components/ScheduleComponent?props=%7B%22amount%22%3A1%2C%22scheduleType%[critical] Uncaught PHP Exception Exception: "Unknown or bad format (P-4D)" at /var/www/html/src/Service/ScheduleExpressionParser.php line 91, referer: http://myaddress:5823/post/create

myaddress - - [24/Oct/2023:00:50:26 +0000] "GET /en/_components/ScheduleComponent?props=%7B%22amount%22%3A1%2C%22scheduleType%22%3A1%2C%22timeZone%22%3A%22America%2FNew_York%22%2C%22weekday%22%3A1%2C%22targetTime%22%3A%2200%3A31%22%2C%22selectedDayType%22%3A0%2C%22day%22%3A1%2C%22%40attributes%22%3A%7B%22data-live-id%22%3A%22live-47055587-0%22%7D%2C%22%40checksum%22%3A%22FuEnv1TKIKsLqLCH%2By0NEfIR8LdDzsBkwYf64QQqJmY%3D%22%7D&updated=%7B%22scheduleType%22%3A%221%22%2C%22weekday%22%3A%225%22%7D HTTP/1.1" 500 1353 "http://myaddress:5823/post/create" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/118.0"

myaddress - - [24/Oct/2023:00:52:46 +0000] "GET /post/list HTTP/1.1" 200 2577 "http://myaddress:5823/post/create" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/118.0"22%3A1%2C%22timeZone%22%3A%22America%2FNew_York%22%2C%22weekday%22%3A1%2C%22targetTime%22%3A%2200%3A31%22%2C%22selectedDayType%22%3A0%2C%22day%22%3A1%2C%22%40attributes%22%3A%7B%22data-live-id%22%3A%22live-47055587-0%22%7D%2C%22%40checksum%22%3A%22FuEnv1TKIKsLqLCH%2By0NEfIR8LdDzsBkwYf64QQqJmY%3D%22%7D&updated=%7B%22scheduleType%22%3A%221%22%2C%22weekday%22%3A%225%22%7D HTTP/1.1" 500 1353 "http://myipaddress:5823/post/create" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/118.0"

myipaddress - - [24/Oct/2023:00:52:46 +0000] "GET /post/list HTTP/1.1" 200 2577 "http://myipaddress:5823/post/create" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/118.0"
```

**Comment by RikudouSage** [comment]
I'll check it out!

**Comment by RikudouSage** [comment]
The issue is entirely unrelated. It's been fixed in #20.
